### ISSUE TITLE
Lightening Media Image: Image Browser Dispaly's id should be image_browser instead of entity_browser for Media View

### ISSUE THREAD

**Issue by purushotamrai** [acquia/lightning#153]
Line No. 163 : lightening_media.install file creates **new display with id entity_browser** for Media View  
Line No. 60 : lightening_media_image.install file creates new display with **same id entity_browser** for Media View

Also, Image Browser config included in Lightening Media Image Feature points to image_browser display of media view while the above approach gives 'entity_browser_2' as display id to that display.

**PR Created to solve this: https://github.com/acquia/lightning/pull/152**

**Comment by phenaproxima** [comment]
So, there is a specific reason why we don't assign the display ID, and allow Views to generate it. The thing is, we cannot assume that the site builder did not already, somehow, have a display with the ID of "image_browser" already in the view. It's highly unlikely, but it's possible, and with something as configuration-dependent as Lightning, we need to consider those sorts of edge cases.

To me, the real problem here is in lightning_media_image.install, around like 549:

```php
  $browser = EntityBrowser::load('image_browser');
  $widgets = $browser->get('widgets');
  if (isset($widgets['58383135-0f34-4a4a-85fc-5cf9b5de2fdd'])) {
    $widgets['58383135-0f34-4a4a-85fc-5cf9b5de2fdd']['settings']['view_display'] = $display_id;
    $browser->set('widgets', $widgets);
    $browser->save();
  }
``` 

It sounds like the real problem here is that the generated display ID is not being saved in the entity browser's configuration, for some reason. To me, the most likely possibility is that there is no widget with the 58383135-0f34-4a4a-85fc-5cf9b5de2fdd UUID -- but, in a clean Lightning install, the image_browser entity browser *will* be created with a widget identified by that UUID. So something is very strange here. IMHO, debugging efforts should focus on figuring out why the widget configuration is not updated, rather than changing the display ID itself.

**Comment by brantwynn** [comment]
> we cannot assume that the site builder did not already, somehow, have a display with the ID of "image_browser" already in the view

Why not just prefix all the display ID's with lightning_ ? Then they can be different, generic AND specific enough to not run over existing configurations.

Honestly, I'm not sure the bit about the widgets is relevant to this thread.  It could be file as its own issue.

**Comment by phenaproxima** [comment]
Prefixing the displays with lightning_ is a pretty good idea, but we'd have to remember to do that in the future rather than apply it retroactively. Creating these displays is basically a one-and-done sort of operation.

That said, I still think that simply changing the display ID is not going to get us to the heart of the matter. Irrespective of whether the ID is generated by Views or set explicitly by Lightning, why did this problem occur? That needs to be answered before we can merge #152.

**Comment by phenaproxima** [comment]
So here's the thing.

Re-reading the OP, I'm not sure this is really what's happening, because Lightning is allowing Views to generate the display IDs automatically. Both are using the entity_browser display plugin, but they are definitely getting unique display IDs, because Views enforces that internally. If Lightning needed to use a specific display ID (rather than auto-detecting it), we would *definitely* set it explicitly in either our exported configuration, or the routines which generate our install-time configuration. So what I'm trying to say is, I'm not sure I see the problem here.

It's possible that this is really just the Views API having poor DX. Consider these lines:

```php
$display_id = $view->addDisplay('entity_browser', 'Image Browser');
```

and

```php
$display_id = $view->addDisplay('entity_browser', 'Browser');
```

The first argument to $view->addDisplay() is the ID of the display plugin, NOT the ID of the display. The display ID is not passed; Views is auto-generating it, which is why we explicitly capture it in $display_id.

In the last 8 or so months since this issue was touched, we haven't seen any other complaints about this, nor anything similar. I think it is safe to close this one.
